toolchain/check/check.cpp

```diff
@@ -950,6 +950,20 @@ static auto CheckParseTree(
 #endif
 }
 
+// Measures duration to produce and check the IR for the provided Parse::Tree.
+static auto CheckParseTree(
+    llvm::MutableArrayRef<Parse::NodeLocConverter> node_converters,
+    UnitInfo& unit_info, int total_ir_count, llvm::raw_ostream* vlog_stream)
+    -> void {
+  auto start_time = std::chrono::steady_clock::now();
+  CheckParseTreeInner(node_converters, unit_info, total_ir_count, vlog_stream);
+  auto end_time = std::chrono::steady_clock::now();
+  auto timings = unit_info.unit->timings;
+  if (timings) {
+    (*timings)->Add("check", end_time - start_time);
+  }
```

Comment:
```suggestion
  if (auto& timings = *(unit_info.unit->timings)) {
    auto end_time = std::chrono::steady_clock::now();
    timings->Add("check", end_time - start_time);
  }
```

This assumes the separate comment about `std::optional<Timings>*` -- the `optional` works in the `if` statement this way, so I think this'll work and feels slightly more idiomatic.